cfbs/commands.py

```diff
@@ -1039,12 +1039,13 @@ def _compare_list(a, b):
         for x, y in zip(a, b):
             if type(x) != type(y):
                 return False
-            if isinstance(x, dict) and not _compare_dict(x, y):
-                return False
-            if isinstance(x, list) and not _compare_list(x, y):
-                return False
-            if x != y:
-                return False
+            if isinstance(x, dict) and _compare_dict(x, y):
+                continue
+            if isinstance(x, list) and _compare_list(x, y):
+                continue
```

Comment:
Are you sure this is the right thing to do? Please extend the commit message with an explanation of how exactly this change fixes the issue.

Reply:
Talked to Lars in person. Although the current approach works, it's probably safer (and more correct) to just do a choice between the 3 options (dict, list, other) and then do the correct comparison inside that block. The current approach with `and` and this kind of "fallthrough" to the logic below, is confusing.

Specifically, it is a bit weird to think about what happens when it is a dict, but `_compare_dict()` returns False, causing us to go down to comparing them again.